core/src/test/java/org/apache/calcite/test/enumerable/EnumerableJoinTest.java

```diff
@@ -258,6 +264,85 @@
         .returnsUnordered("empid=100\nempid=110\nempid=150\nempid=200");
   }
 
+  private static final SqlCollation NO_CASE =
+      new SqlCollation(CalciteSystemProperty.DEFAULT_COLLATION.value() + "-NO_CASE",
+          SqlCollation.Coercibility.IMPLICIT) {
+        @Override public Comparator getComparator() {
+          return String.CASE_INSENSITIVE_ORDER;
+        }
+        @Override public Expression getComparatorExpression() {
+          return Expressions.field(Expressions.constant("", String.class),
```

Comment:
Not sure if there is a way to express `String.CASE_INSENSITIVE_ORDER` as an Expression... (for the moment using `"".CASE_INSENSITIVE_ORDER` as workaround)